### PR TITLE
Merging internal 2021 Release Wave 2 fixes

### PR DESCRIPTION
**Pull request by TYLEROL**
### Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Other (updates to documentation, formatting, etc.)

### Description
- Fixes for Microsoft Dataverse 2021 Release Wave 2 update

### Issues addressed

- Fix issue with save due to language specific labels
- Addressed an issue with the ViewPicker
- Updates to address new grid control format
- Addressed an issue with personal settings flyout

### All submissions:

- [x] My code follows the code style of this project.
- [x] Do existing samples that are effected by this change still run?
- [x] I have added samples for new functionality. 
- [x] I raise detailed error messages when possible.
- [x] My code does not rely on labels that have the option to be hidden.

### Which browsers was this tested on?

- [x] Chrome
- [ ] Firefox
- [ ] IE
- [ ] Edge

### CODE REVIEW

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs [hunk 1]
```diff
@@ -1488,84 +1524,67 @@ internal BrowserCommandResult<bool> SwitchView(string viewName, string subViewNa
...
-
-                driver.WaitForTransaction();
-
-                return true;
-            });
+            Actions actions = new Actions(driver);
+            var CurrentRows = driver.FindElements(By.XPath(AppElements.Xpath[AppReference.Grid.Rows]));
+            var lastFloor = CurrentRows.Where(x => Convert.ToInt32(x.GetAttribute("row-index")) == lastKnownFloor).First();
+            var topRow = driver.FindElement(By.XPath(AppElements.Xpath[AppReference.Grid.Rows]));
+            var firstCell = lastFloor.FindElement(By.XPath("//div[@aria-colindex='1']"));
```
Comment (dtu11) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs:1533]: Can we move the xpath to the references file?

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs [hunk 2]
```diff
@@ -1488,84 +1524,67 @@ internal BrowserCommandResult<bool> SwitchView(string viewName, string subViewNa
...
+                        lastRow = true;
+                    }
+                    if (driver.HasElement(By.XPath(AppElements.Xpath[AppReference.Grid.LastRow])))
+                    {
+                        lastRow = true;
+                    }
                 }
+                if (gridRow == null) throw new NotFoundException($"Grid Row {index} not found.");
+                var xpathToGrid = By.XPath("//div[contains(@data-lp-id,'MscrmControls.Grid')]");
+                var gridContainer = driver.FindElement(By.XPath("//div[contains(@data-lp-id,'MscrmControls.Grid')]"));
```
Comment (dtu11) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs:1567]: Move xpath to references file

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs [hunk 3]
```diff
@@ -1605,53 +1624,105 @@ internal BrowserCommandResult<bool> ClearSearch(int thinkTime = Constants.Defaul
...
+            }
+        }
         internal BrowserCommandResult<List<GridItem>> GetGridItems(int thinkTime = Constants.DefaultThinkTime)
         {
             ThinkTime(thinkTime);
 
             return this.Execute(GetOptions("Get Grid Items"), driver =>
             {
                 var returnList = new List<GridItem>();
+                var gridContainer = driver.FindElement(By.XPath("//div[contains(@data-lp-id,'MscrmControls.Grid')]"));
```
Comment (dtu11) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs:1691]: Move xpath to references file

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs [hunk 4]
```diff
@@ -1849,17 +1920,17 @@ public BrowserCommandResult<bool> Sort(string columnName, string sortOptionButto
...
+                var sortCol = driver.FindElement(By.XPath(AppElements.Xpath[AppReference.Grid.GridSortColumn].Replace("[COLNAME]", columnName)));
 
-                column.Click(true);
-                driver.WaitForTransaction();
+                if (sortCol == null)
+                    throw new InvalidOperationException($"Column: {columnName} Does not exist");
+                else
+                {
+                    sortCol.Click(true);
+                    driver.WaitUntilClickable(By.XPath($@"//button[@name='{sortOptionButtonText}']")).Click(true);
```
Comment (dtu11) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs:1930]: Move xpath to references file

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs [hunk 5]
```diff
@@ -2170,6 +2289,311 @@ internal BrowserCommandResult<bool> SearchSubGrid(string subGridName, string sea
...
+
+                                    lastRow = true;
+                                }
+                                else
+                                {
+                                    actions = ClickSubGridAndPageDown(driver, grid);
+                                }
+
+                            }
+                            if (grid.HasElement(By.XPath("//div[@data-control-name='Subgrid_1']//div[@ref='centerContainer']//div[@role='rowgroup']//div[@role='row' and contains(@class, 'ag-row-last')]")))
```
Comment (dtu11) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs:2534]: Move xpath to references file

File: README.md [hunk 6]
```diff
@@ -54,7 +52,6 @@ Selenium is a set of different software tools each with a different approach to
 Although we don't have specific commands to cover the above funcationality, we have generic commands that will allow you to still interact with those features. 
 
 ## Known Issues #
-- Microsoft Edge Browser, based on Chromium, is currently not supported.
```
Comment (Shakevg) [inline README.md:57]: Does Edge Chromium support now?

**Pull request merged**